src/material/schematics/ng-add/setup-project.ts

```diff
@@ -33,13 +33,25 @@ const noopAnimationsModuleName = 'NoopAnimationsModule';
  *  - Adds Browser Animation to app.module
  */
 export default function(options: Schema): Rule {
-  return chain([
-    addAnimationsModule(options),
-    addThemeToAppStyles(options),
-    addFontsToIndex(options),
-    addMaterialAppStyles(options),
-    addTypographyClass(options),
-  ]);
+  return (host: Tree, context: SchematicContext) => {
+    const workspace = getWorkspace(host);
+    const project = getProjectFromWorkspace(workspace, options.project);
+
+    if (project.projectType === 'application') {
+      return chain([
+        addAnimationsModule(options),
+        addThemeToAppStyles(options),
+        addFontsToIndex(options),
+        addMaterialAppStyles(options),
+        addTypographyClass(options),
+      ]);
+    }
+    context.logger.warn(
+        'Unable to set up Angular Material in a library project as there is no special ' +
+        'setup required. If you intended to run the schematic for a different project, ' +
+        'pass the `--project` option.');
```

Comment:
I just came across this again. I think we need to update the message slightly. If library authors run `ng add @angular/material`, then they probably only expect us to add the dependency. This is done by the CLI already, so the message that we're unable to setup Angular Material might be misleading.

Can we rephrase is so that we just explain that there is no special setup needed for libraries?

Reply:
@devversion something like
`Angular Material has been added to your library project dependencies. This kind of projects does not require any further setup. If you intended to run the schematic for a different project, pass the `--project` option.`

Reply:
I'm sorry. I wanted to get back to you as soon as possible, but so many other things were going on and I forgot. 

I'd propose something like:

```
Angular Material has been set up in your workspace. There is no additional
setup required for consuming Angular Material in your library project.

If you intended to run the schematic on a different project, pass the `--project` option.
```

Reply:
@devversion no worries. updated and rebased

Reply:
Awesome, thx for addressing! Looks like there is a test failure. I'll add merge ready when that is solved.